__tests__/integration/hathorwallet_facade.test.js

```diff
@@ -0,0 +1,1665 @@
+    expect(sendTx.inputs).toHaveLength(2);
+    expect(sendTx.outputs).toHaveLength(4);
+
+    // Validating that each of the outputs has the values we expect
+    const smallerHtrOutput = sendTx.outputs.find(o => o.value === 3);
+    expect(smallerHtrOutput).toBeDefined();
+    const biggerHtrOutput = sendTx.outputs.find(o => o.value === 5);
+    expect(biggerHtrOutput).toBeDefined();
+    const smallerTokenOutput = sendTx.outputs.find(o => o.value === 90);
+    expect(smallerTokenOutput.token).toEqual(tokenUid);
+    const biggerTokenOutput = sendTx.outputs.find(o => o.value === 110);
+    expect(biggerTokenOutput.token).toEqual(tokenUid);
+
+    // Validating that each of the inputs has the values we expect
+    const htrInput = sendTx.inputs.find(o => o.token === HATHOR_TOKEN_CONFIG.uid);
+    expect(htrInput.value).toEqual(8);
+    const tokenInput = sendTx.inputs.find(o => o.token === tokenUid);
+    expect(tokenInput.value).toEqual(200);
+  });
+
+  it('should respect timelocks', async () => {
+    const hWallet = await generateWalletHelper();
+    await GenesisWalletHelper.injectFunds(hWallet.getAddressAtIndex(0), 10);
+
+    // Defining timelocks (milliseconds) and timestamps (seconds)
+    const startTime = Date.now().valueOf();
+    const timelock1 = startTime + 5000; // 5 seconds of locked resources
+    const timelock2 = startTime + 8000; // 8 seconds of locked resources
+    const timelock1Timestamp = dateFormatter.dateToTimestamp(new Date(timelock1));
+    const timelock2Timestamp = dateFormatter.dateToTimestamp(new Date(timelock2));
+
+    const rawTimelockTx = await hWallet.sendManyOutputsTransaction(
+      [
+        {
+          address: hWallet.getAddressAtIndex(1),
+          value: 7,
+          token: HATHOR_TOKEN_CONFIG.uid,
+          timelock: timelock1Timestamp,
+        },
+        {
+          address: hWallet.getAddressAtIndex(1),
+          value: 3,
+          token: HATHOR_TOKEN_CONFIG.uid,
+          timelock: timelock2Timestamp,
+        }
+      ],
+    );
+    await waitForTxReceived(hWallet, rawTimelockTx.hash);
+
+    // Validating the transaction with getFullHistory / getTx
+    const timelockTx = hWallet.getTx(rawTimelockTx.hash);
+    expect(timelockTx.outputs.find(o => o.decoded.timelock === timelock1Timestamp)).toBeDefined();
+    expect(timelockTx.outputs.find(o => o.decoded.timelock === timelock2Timestamp)).toBeDefined();
+
+    // Validating getBalance ( moment 0 )
+    let htrBalance = await hWallet.getBalance(HATHOR_TOKEN_CONFIG.uid);
+    expect(htrBalance[0].balance.locked).toBe(10);
+    expect(htrBalance[0].balance.unlocked).toBe(0);
+
+    // Validating interfaces with only a partial lock of the resources
+    const waitFor1 = timelock1 - Date.now().valueOf() + 1000;
+    loggers.test.log(`Will wait for ${waitFor1}ms for timelock1 to expire`);
+    await delay(waitFor1);
+
+    /*
+     * The locked/unlocked balances are usually updated when new transactions arrive.
```

Comment:
That's also a weird behavior, I thought it would be available when the time comes in this facade. We should create an issue to investigate that.

Reply:
Did we open this issue?

Reply:
Opened this issue to investigate:
- #398